portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -144,6 +126,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.getWriter().println(gson.toJson(LoadAllBlogsOrLast(/*all=*/false)));   
   }
  
+  private List<BlogMessage> putRepliesWithPosts(List<BlogMessage> blogMessages) {
+    
+ 
+    // TODO: Use helper functions for lines 73-101.
+ 
+    // Separate posts from replies from |blogMessages|.
+    // Add replies to messageReplies for the respective posts.
+    List<BlogMessage> blogMessagesReplies = new ArrayList<BlogMessage>();
+    for (BlogMessage message : blogMessages) {
+      if (message.getParentID() != 0) {
+        blogMessagesReplies.add(message);
+        blogMessages.remove(message);
+      }
+    }
+ 
+    for (BlogMessage post : blogMessages) {
+      for (BlogMessage reply : blogMessagesReplies) {
+        if (reply.getParentID() == post.getTimestamp()) {
+          post.addReply(reply);
```

Comment:
Similarly to Andrew's "removing while iterating" comment, you need to be very careful when you change the container when you iterate through it. Usually you would want to either avoid that or use explicitly iterator.

Reply:
+1. but in this case i think the container itself is not changing? post is not removed or added: one of its members, 'messageReplies' is getting another item 'reply'. so the iteration container doesn't change i think?

Reply:
Yeah, since reply is being added to the list of replies for the post it shouldn't change the number of items in blogMessages.